### DISCUSSION TITLE
Efficient random moves - not all legal moves

### DISCUSSION THREAD

**Discussion by 33fred33** [niklasf/python-chess#1042] [Q&A]
I am writing search algorithms for chess. For this, I need to run random games. As of now and as far as I know, the only way to get a random move with this library is to:
1 Get the list of legal moves
2 Choose one at random

It would be very handy to have a function that returns a random legal move without the need to calculate all the legal moves available
Is there a way I am not aware of?

**Comment by niklasf** [comment]
To generate moves uniformly at random, legality of all candidates has to be checked, which is not better than generating a list of all legal moves.

If the distribution does not matter, one strategy could be to generate moves only for a randomly selected piece or region of the board. This is possible using `board.generate_legal_moves(from_mask=...)`. If no legal moves are found for the mask, all moves could be generated as a fallback.
